django_usaid_clinicfinder/clinicfinder/tasks.py

```diff
@@ -0,0 +1,197 @@
+                    password=settings.LBS_API_PASSWORD,
+                    msisdn=lbsrequest.search["msisdn"])
+                if result[0][0]["_code"] != "101":
+                    l.info("Failed to return location")
+                    lbsrequest.response = {
+                        "lookup_message": result[0][0]["_message"]
+                    }
+                    lbsrequest.save()
+                    response = result[0][0]["_message"]
+                else:
+                    l.info("Location found, creating lookup")
+                    lbsrequest.response = {
+                        "x": result[0][0]["x"],
+                        "y": result[0][0]["y"],
+                        "lookup_message": result[0][0]["_message"]
+                    }
+                    lbsrequest.save()
+                    # Create location point
+                    location = LookupLocation()
+                    location.point = Point(
+                        float(result[0][0]["x"]), float(result[0][0]["y"]))
+                    location.save()
+                    # set the location object for POI to the location
+                    lookup_poi = lbsrequest.pointofinterest
+                    lookup_poi.location = location
+                    lookup_poi.save()
+                    l.info("Location and Point of Interest linked")
+                    response = result[0][0]["_message"]
+            return response
+        except SoftTimeLimitExceeded:
+            logger.error(
+                'Soft time limit exceed processing LBS lookup through \
+                SOAP API via Celery.',
+                exc_info=True)
+
+lbs_lookup = LBS_Lookup()
+
+
+class Location_Sender(Task):
+
+    """
+    Task to take results and send them off. SMS only for now.
+    """
+    name = "clinicfinder.tasks.location_sender"
+
+    class FailedEventRequest(Exception):
+
+        """
+        The attempted task failed because of a non-200 HTTP return
+        code.
+        """
+
+    def run(self, lookuppointofinterest_id, **kwargs):
+        """
+        Returns a filtered list of locations for query
+        """
+        l = self.get_logger(**kwargs)
+
+        l.info("Processing new location result sending")
+        try:
+            lookuppoi = LookupPointOfInterest.objects.get(
+                pk=lookuppointofinterest_id)
+            response = lookuppoi.response
+            if response["type"] == "SMS" and "sent" not in response:
+                # send via Vumi
+                sender = HttpApiSender(
+                    account_key=settings.VUMI_GO_ACCOUNT_KEY,
+                    conversation_key=settings.VUMI_GO_CONVERSATION_KEY,
+                    conversation_token=settings.VUMI_GO_ACCOUNT_TOKEN
+                )
+                content = response["template"].replace(
+                    "{{ results }}", response["results"])
+                vumiresponse = sender.send_text(response["to_addr"], content)
```

Comment:
Wonder if it's worthwhile having a sanity check on the length of the SMS content here? Don't want to accidentally send someone 3k of text?